src/calendar-heatmap.js

```diff
@@ -111,31 +141,78 @@ function calendarHeatmap() {
         });
       }
 
+      if (dayNumbersInBox){
+        dayNumbers = svg.selectAll('.day-numbers')
+        .data(dateRange);  //  array of days for the last yr
+        
+        dayNumbers.enter().append('text')
+        .attr('class', 'day-numbers')
+        .attr("font-size", dayNumbersFontSz+"px")
+        .attr('x', function (d, i) {
+          var cellDate = moment(d);
+          var numSzBlank = 0;
+          if(d.getDate()>9){
+            numSzBlank = 1; 
+          }else{
+            numSzBlank = 3;
+          }
+          var monthSpacing =  0;
+          if (monthSpace){
+            monthSpacing = cellDate.diff((firstDate).startOf('month'), 'months');
+          }
+          var result = (cellDate.week() + monthSpacing) - firstDate.week() + (firstDate.weeksInYear() * (cellDate.weekYear() - firstDate.weekYear()));
+          return (result * (SQUARE_LENGTH + SQUARE_PADDING)) + (((SQUARE_LENGTH-dayNumbersFontSz)/4) * (numSzBlank));
+        })
+        .attr('y', function (d, i) { return MONTH_LABEL_PADDING + dayNumbersFontSz + d.getDay() * (SQUARE_LENGTH + SQUARE_PADDING); })
+        .text(function (d) { return d.getDate() ; });
+      }
+
       if (chart.tooltipEnabled()) {
         dayRects.on('mouseover', function (d, i) {
+          var monthSpacing =  0;
+            if (monthSpace){
+              monthSpacing = moment(d).diff((firstDate).startOf('month'), 'months');
+            }
           tooltip = d3.select(chart.selector())
             .append('div')
             .attr('class', 'day-cell-tooltip')
             .html(tooltipHTMLForDate(d))
-            .style('left', function () { return Math.floor(i / 7) * SQUARE_LENGTH + 'px'; })
-            .style('top', function () { return d.getDay() * (SQUARE_LENGTH + SQUARE_PADDING) + MONTH_LABEL_PADDING * 3 + 'px'; });
+            .style('left', function () { return (Math.floor(i / 7)+monthSpacing) * SQUARE_LENGTH + 'px'; })
+            .style('top', function () { return d.getDay() * (SQUARE_LENGTH + SQUARE_PADDING) + (MONTH_LABEL_PADDING - 5) * 3 + 'px'; });
         })
         .on('mouseout', function (d, i) {
           tooltip.remove();
         });
+        if (dayNumbersInBox){
+          dayNumbers.on('mouseover', function (d, i) {
```

Comment:
I have little experience on this subject or, cant understand.. Please change as required.